src/main/java/com/google/cloud/genomics/dataflow/readers/bam/BAMShard.java

```diff
@@ -106,7 +106,6 @@ public long approximateSizeInBytes() {
 
   @Override
   public String toString() {
-    return file + ": " + contig.toString() + ", locus size = " + sizeInLoci() + 
-        ", span size = " + approximateSizeInBytes();
```

Comment:
Can you change it to use cached value instead and if cached value not set then don't output it ?
That way if the computation was used in some decision making then cachedvalue is not zero and will be outputted and if not then nobody cares about it.